src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,95 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (newMovie: Movie) => void;
+};
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrlt] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
```

Comment:
It's better to use a single state object for all values, simplifies state management and reduces code complexity. It also minimizes unnecessary re-renders, as updating multiple values in one object triggers only one re-render.
```suggestion
const [formData, setFormData] = useState({
  description: '',
  title: '',
  imgUrl: '',
  imdbUrl: '',
  imdbId: ''
});
```